pcsx2/Gif.cpp

```diff
@@ -194,12 +311,15 @@ static __fi tDMA_TAG* ReadTag2()
 
 bool CheckPaths(EE_EventType Channel) {
 	// Can't do Path 3, so try dma again later...
-	if(!gifUnit.CanDoPath3()) {
-		if(!gifUnit.Path3Masked())
-		{
-			CPU_INT(Channel, 128);
+	if (!CHECK_GIFFIFOHACK) {
+		if (!gifUnit.CanDoPath3()) {
+			if (!gifUnit.Path3Masked())
+			{
```

Comment:
Oh you're right,  I completely missed that return. why not just do an early return if ``CHECK_GIFFIFOHACK`` is a true value ? That'd make the code more simpler.